lib/handlers/compile.ts

```diff
@@ -553,7 +553,9 @@ export class CompileHandler {
                     }
                 },
                 error => {
-                    if (typeof error !== 'string') {
+                    if (typeof error === 'string') {
```

Comment:
Thinking through this more, is there a way we can turn off this rule? (I assume it's a new eslint rule)

Personally I'd far prefer the error case cold path second:
```ts
if(!c) {
    ...
    ...
    the actual important logic
    ...
    ...
} else {
    throw "Error";
}
```

Reply:
Happy to turn this rule off, it does seem quite chatty here. I like it in general, but it's quite heavy-handed.

Reply:
now to find out what the rule's name is though

Reply:
I'm not sure in this case that's the cold case though? This one is handling an error of either string or object type?